### PR TITLE
Review

### PR DESCRIPTION
**Pull request by AVS18**
Added a code function that checks whether a URL shortcode has been already created. If yes then Rerun the same loop to create a new shortcode.

 
let check = getShortCodeDetails(shortCode)    
if(check!==undefined) {
        return await createRandomShortCode(longUrl);    
}

### CODE REVIEW

**Comment by championswimmer** [comment]
Please name the pull request as `Fixes <xyz> issue`, or `Adds <xyz> feature` to be more descriptive :)

File: src/controllers/shortcode.controller.ts [hunk 1]
```diff
@@ -8,6 +8,10 @@ export async function createRandomShortCode(longUrl: string): Promise<ShortCode>
     const shortCode = intToRadix64(randomCode)
 
     // TODO: retry if exists
+    let check = getShortCodeDetails(shortCode)
+    if(check!==undefined){
```
Comment (championswimmer) [inline src/controllers/shortcode.controller.ts:12]: if (!check) is a better check as that checks for `false`, `null`, `undefined` all 3 cases. 

Also your if-condition is opposite. We should recurse only if it **IS** undefined, not the other way around.

**Review by championswimmer — CHANGES_REQUESTED** [review]

File: src/controllers/shortcode.controller.ts (hunk 1 above)
Comment (AVS18) [reply src/controllers/shortcode.controller.ts:12]: if(!check) I will add. Can you elaborate more over the second one that recurse only if it IS undefined.
as of now, I did check for the shortcode if exist and if it is not undefined that means the shortcode already exist in the database hence you need to again recall to generate a new code.

Comment (championswimmer) [reply src/controllers/shortcode.controller.ts:12]: ah ok my bad. logic was correct in yours. 

btw you need to await the response of  `getShortCodeDetails(shortCode)`

Comment (AVS18) [reply src/controllers/shortcode.controller.ts:12]: Hi, i missed some conversation here. Is this still unresolved.

**Pull request closed without merging**